compiler/IREmitter/IREmitterHelpers.cc

```diff
@@ -236,11 +221,26 @@ void IREmitterHelpers::emitReturn(CompilerState &cs, llvm::IRBuilderBase &build,
     }
 }
 
-void IREmitterHelpers::setThrowReturnFlag(CompilerState &cs, llvm::IRBuilderBase &build, const IREmitterContext &irctx,
-                                          int rubyBlockId) {
-    auto &builder = static_cast<llvm::IRBuilder<> &>(build);
+llvm::Value *IREmitterHelpers::maybeCheckReturnValue(CompilerState &cs, cfg::CFG &cfg, llvm::IRBuilderBase &build,
```

Comment:
The fact that this can return `voidSingleton` seems elegant to me, but it wasn't clear at the call site. Kinda wonder if that suggest that this function could use a better name, but can't think of anything so :shrug:

Reply:
Yeah, I don't know what a better name for this function is.  `maybeTypecheckOrAdjustReturnValue`?  That's a mouthful.

Reply:
Indeed. As it is I suppose most readers will probably notice at the call site that it's returning an `llvm::Value *`, wonder why, and chase this down anyway. (So maybe a comment would help here, but it's up to you.)

Reply:
I added a comment at the declaration.